documentation: ZENKO-1480_OOB_updates_for_SOFS

wabernat opened the pull request:
This PR updates the documentation as concerns the SOFS/NFS feature

This PR introduced the following changes: 

- Updated location creation process.
- Fixed a duplicated image (bad link) in System Architecture.
- Rationalized file titles in Installation Guide.
- Enhanced upgrade procedures.
- Added "Supported Protocols".
- Renamed Zenko_Architecture directory (Now "Architecture").
- Renamed index files as appropriate.
- Updated CIFS and NFS installation sections.

This fixes ZENKO-1480 (and quite a few extraneous issues revealed when addressing ZENKO-1480)

bert-e commented:
# Hello wabernatscality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/zenko/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-1480 contains:


* `Zenko PSI 1.1 `


Considering where you are trying to merge, I expected to find:


* `1.1.0`


Please check the `Fix Version/s` of ZENKO-1480, or the target
branch of this pull request.

wabernat commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @giacomoguiulfo

* @ssalaues

* @nicolast

* @vrancurel

wabernat commented:
@bert-e approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @giacomoguiulfo

* @ssalaues

* @nicolast

* @vrancurel










*The following options are set:* **approve**

ghost commented:
There were the following issues with this Pull Request

* Commit: bada20e25f5b126b6745a117fcb73bef21c37672
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

ssalaues commented on docs/docsource/Installation/configure/configuring_zenko.rst:
```diff
@@ -0,0 +1,90 @@
+      written to Zenko/kubernetes/zenko/charts/grafana/values.yaml).
+   #. Base settings for Zenko. These settings override the base microservice 
+      settings, and are found in Zenko/kubernetes/zenko/values.yaml.
+   #. Custom settings, which you can write to an options.yaml chart. Settings
```
The `options.yaml chart` seems misleading. I would omit the `chart` or replace it with `file`

ssalaues commented on docs/docsource/Installation/configure/configuring_zenko.rst:
```diff
@@ -0,0 +1,90 @@
+   #. Base settings for Zenko. These settings override the base microservice 
+      settings, and are found in Zenko/kubernetes/zenko/values.yaml.
+   #. Custom settings, which you can write to an options.yaml chart. Settings
+      written to this chart override settings read from the preceding
```
Same here regarding the use of `chart` in this case it's just a yaml file that we call 'options.yaml' for the sake of example.

ssalaues commented on docs/docsource/Installation/configure/configuring_zenko.rst:
```diff
@@ -0,0 +1,90 @@
+Modifying options.yaml
+----------------------
+
+The options.yml chart is not installed by default, but you added a simple one
```
Couple uses of `options.yaml chart` in this paragraph but otherwise really good info on why we recommend using options files.

ssalaues commented on docs/docsource/Installation/configure/configuring_zenko.rst:
```diff
@@ -0,0 +1,90 @@
+nodeCount parameter, issue the following command from Zenko/kubernetes/zenko
+to push your changes to the deployed Zenko instance::
+
+   $ helm upgrade {{zenko-server-name}} .
```
this is missing the `-f options.yaml`

ssalaues reviewed (changes requested):
Maybe it's mis-communication from my part but mostly just some comments about the use of the phrase `options.yaml chart`.

bert-e commented:
# Conflict




There is a conflict between your branch `documentation/ZENKO-1480_OOB_updates_for_SOFS` and the
destination branch `development/1.1`.

Please resolve the conflict on **the feature branch** (`documentation/ZENKO-1480_OOB_updates_for_SOFS`).

```bash
 $ git fetch
 $ git checkout origin/documentation/ZENKO-1480_OOB_updates_for_SOFS
 $ git merge origin/development/1.1
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:documentation/ZENKO-1480_OOB_updates_for_SOFS
```







*The following options are set:* **approve**

ghost commented:
There were the following issues with this Pull Request

* Commit: bee0da1ff9e30070abea8cccece1f2e31a8f2b47
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

ghost commented:
There were the following issues with this Pull Request

* Commit: bee0da1ff9e30070abea8cccece1f2e31a8f2b47
  - ✖ message may not be empty
  - ✖ type may not be empty
* Commit: bb95a5cdee250c3040bd396d493499406a34c441
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @tmacro

* @nicolast

* @ssalaues

* @vrancurel





The following reviewers are expecting changes from the author, or must review again:

* @ssalaues







*The following options are set:* **approve**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @tmacro

* @nicolast

* @ssalaues

* @vrancurel










*The following options are set:* **approve**

anurag4DSB commented on docs/docsource/Installation/configure/configuring_zenko.rst:
```diff
@@ -0,0 +1,90 @@
+Modifying options.yaml
+----------------------
+
+The options.yml file is not installed by default, but you added a simple one
```
Suggestion: The options.yml file is **not present** by default, but you **can add** a simple one

wabernat replied:
Welll...This is a reference to an earlier step. I can accept "not present," but they had to have added one (or at the very least *should* have added one) to have gotten this far. 

Either way, I need to fix this, because options.yml does not have an "a" in its suffix!

anurag4DSB commented on docs/docsource/Operation-Architecture/Orbit_UI/Location_Management/Adding_a_Storage_Location.rst:
```diff
@@ -11,31 +11,36 @@ Storage Locations dialog:
-    this cap is planned for subsequent product revisions.
-
+    Google Cloud Platform imposes a 1024-object cap. For all other 
+    backends, Zenko caps the number of objects at 10,000.
```
Is this for open source version or something?

wabernat replied:
The big problem here is that this should say "part" not "object".

As I understand it, these are architectural limits, though as Alex clarifies, the 1024-part cap comes from us, not GCP. The S3 protocol caps us at 10,000 parts.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @tmacro

* @nicolast

* @ssalaues

* @vrancurel





The following reviewers are expecting changes from the author, or must review again:

* @anurag4dsb







*The following options are set:* **approve**

wabernat committed:
documentation: ZENKO-1480_OOB_updates_for_SOFS

Updated location creation process.
Fixed a duplicated image (bad link) in System Architecture.
Rationalized file titles in Installation Guide.
Enhanced upgrade procedures.
Added "Supported Protocols".
Renamed Zenko_Architecture directory (Now "Architecture").
Renamed index files as appropriate.
Updated CIFS and NFS installation sections.

Changed "charts" to "files" or deleted

added -f options.yml

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-1480 contains:


* `Zenko 1.1 GA`

* `Zenko 1.1 RC2`


Considering where you are trying to merge, I expected to find:


* `1.1.0`


Please check the `Fix Version/s` of ZENKO-1480, or the target
branch of this pull request.




*The following options are set:* **approve**

wabernat commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/Zenko/commit/9ebb3e414d27a491290450f25640f3317b781368) did not succeed in branch documentation/ZENKO-1480_OOB_updates_for_SOFS.




*The following options are set:* **approve**

ssalaues commented:
@bert-e bypass_build_status

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/1.1`



The following branches will **NOT** be impacted:

* `development/0.9`
* `development/1.0`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **bypass_build_status, approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/1.1`



The following branches have **NOT** changed:

* `development/0.9`
* `development/1.0`



Please check the status of the associated issue ZENKO-1480.

Goodbye wabernatscality.
